### PR TITLE
feature(file): more consistency in mime and simple type values

### LINKED ISSUE #9614: Do not rely on metadata values for mime and simple type
I propose we move away from relying on metadata for mime and simple type values.  It doesn't cost us much to parse these, but it would offer consistency in handling of files. It would also eliminate errors due to omissions by plugins.

### PR DESCRIPTION
**Pull request by hypeJunction**
Adds ElggFile::getSimpleType and patches ElggFile::getMimeType to parse type values
if not set as metadata.

Fixes #9614

### CODE REVIEW

**Comment by mrclay** [comment]
👍 for moving more tests to PHPUnit. LGTM.

**Comment by hypeJunction** [comment]
Not sure if we should populate metadata on save. Makes sense on one hand,
might lead to surprises on the other hand.

**Comment by mrclay** [comment]
Agreed, remove the auto-populate metadata on save. Can always be added later.

**Comment by hypeJunction** [comment]
@mrclay Removed metadata save

File: engine/classes/ElggFile.php [hunk 1]
```diff
@@ -172,6 +173,19 @@ public function detectMimeType($file = null, $default = null) {
 	}
 
 	/**
+	 * Get the simple type of the file.
```
Comment (mrclay) [inline engine/classes/ElggFile.php]: Can you give example returns here, or add a `@see elgg_get_file_simple_type`

Comment (mrclay) [inline engine/classes/ElggFile.php]: LGTM

Comment (hypeJunction) [inline engine/classes/ElggFile.php]: Added

**Pull request merged**